api/types/provisioning.go

```diff
@@ -410,3 +426,23 @@ func (a *ProvisionTokenSpecV2GitHub) checkAndSetDefaults() error {
 	}
 	return nil
 }
+
+func (a *ProvisionTokenSpecV2CircleCI) checkAndSetDefaults() error {
+	if len(a.Allow) == 0 {
+		return trace.BadParameter("the %q join method requires at least one token allow rule", JoinMethodCircleCI)
+	}
+	if a.OrganizationID == "" {
+		return trace.BadParameter("the %q join method requires 'organization_id' to be set", JoinMethodCircleCI)
+	}
+	for _, rule := range a.Allow {
+		projectSet := rule.ProjectID != ""
+		contextSet := rule.ContextID != ""
+		if !(projectSet || contextSet) {
```

Comment:
Are project and context both required? The logic here says yes, but the error message makes it sound like only one is required.

Reply:
Hmm, I'm pretty sure this behaves as the error describes (as in, only a single one must be set for it to be valid). I could rewrite it as:
```
if (!projectSet && !contextSet)
```

If you think that would be more understandable.

Reply:
Yeah, that's a bit easier for me to reason about. (You're right by the way, I had confused myself on the first review)

Reply:
d12f98df97